resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -44,7 +44,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val EXECUTOR_MODIFICATION_LOCK = new Object
   private val runningExecutorPods = new scala.collection.mutable.HashMap[String, Pod]
 
-  private val kubernetesMaster = "https://kubernetes"
+  private val kubernetesMaster = "https://kubernetes.default.svc"
```

Comment:
Yup this works on my machine. Good find!

I think this string belongs in `constants.scala` and `KubernetesClientBuilder#buildFromWithinPod()` can just use that string directly. Remove the `kubernetesMaster` parameter from that method.